src/diagnostics.rs

```diff
@@ -0,0 +1,461 @@
+//! The diagnostics object controls the output of warnings and errors generated
+//! by the compiler during the lexing, parsing and semantic analysis phases.
+//! It also tracks the number of warnings and errors generated for flow control.
+//!
+//! This implementation is NOT thread-safe.
```

Comment:
Isn't this redundant given that `RefCell` is `!Sync` and thus any `#[derive(Send,Sync)]` will fail anyways?